app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -162,10 +167,41 @@ def os_disk_caching
       @azure_vm.storage_profile.os_disk.caching
     end
 
+    def image_id
+      image = @azure_vm.storage_profile.image_reference
+      return nil unless image
+      image_uuid = "#{image.publisher}:#{image.offer}:#{image.sku}:#{image.version}"
+      image_uuid
+    end
+
+    def vm_extension
+      return nil unless @azure_vm.resources
+      @vm_extension ||= begin
+        ext_name = @azure_vm.resources.first.id.split('/')[-1]
```

Comment:
Not sure but would it be possible where `VM` don't have any resources?
If yes then it would be good to handle that condition too.

Reply:
It would only have **no resources** in case of `Compute Profile attributes` page and the first time call to `new_vm` when all args are empty.

Reply:
yes, I have seen nil handling.
I would like to know - is there any chance that provisioned vm don't have any resource i.e vm_extension? so that it will returned empty array in resources.

Because on line 180, you have directly called `@azure_vm.resources.first.id`, if no extension available then it will again raise an error.

Reply:
A provisioned VM will always have `@azure_vm.resources` value present. Even if script command/uris are empty.